### ISSUE TITLE
[BUG][FormTribe 🔥][100 Points] Multiple Choice Question's Other Field submits each value onChange rather than the final Value

### ISSUE THREAD

**Issue by ShubhamPalriwala** [formbricks/formbricks#1028]
### Welcome to the FormTribe

This issue will get you the points mentioned in issue title for the [FormTribe Hackathon](https://formtribe.com)

**Preparation:**
If this is your first time contributing, please read:

- Our [introductory blog post](https://formbricks.com/blog/join-the-formtribe)
- Overview of [Code Guidelines at Formbricks.](https://formbricks.com/docs/contributing/how-we-code)

**Only when your PR follows _all_ of the best practices it will get merged.**

Want more points quick? Complete [Side Quests 🏰 ](https://formbricks.com/formtribe#side-quests)

--------

### Issue Summary

In the Multiple Choice question that has an option for an "Other" field as well, type any value in the other box and hit submit. In the survey responses, you will see that instead of just getting 1 final value, you keep on getting all the values as soon as the user makes a change to the input value ie (adds a letter, a space, a backspace, etc).

Here's a loom video walking you through the same:
https://www.loom.com/share/21a9a190ad874a8394e40ad3ee0f51cc?sid=2845b8b1-1bd0-48e1-bff4-6a451790f72e

### Steps to Reproduce

1. Create a Survey with Multiple Choice question and add an "Other" option type and Publish the survey
2. Now open the survey, click on the other option and fill anything and click submit
3. Now go to the survey response page and view how the response is shown.

### Expected behavior

Instead of each value the user is typing, we should only get the final value and show that.

### Other information

Look into the MultipleChoiceMultipleQuestion component we have [here](https://github.com/formbricks/formbricks/blob/9bab07077814a8b44ab31fd252832d18cf01dd5f/packages/surveys/src/components/MultipleChoiceMultiQuestion.tsx)

<details>
  <summary>If you want a hint to get closer to the fix (Take it as a challenge and do not open the below before you fix the bug)</summary>
This might have something to what you're looking for:
https://github.com/formbricks/formbricks/blob/main/packages/surveys/src/components/MultipleChoiceMultiQuestion.tsx#L178

</details>

### Environment

- [X] Formbricks Cloud (app.formbricks.com)
- [X] Self-hosted Formbricks

### Desktop (please complete the following information)

```markdown
- OS:
- Node:
- npm:
```

**Comment by prasoonmhwr** [comment]
May I work on this

**Comment by ShubhamPalriwala** [comment]
thank you 🚀 , its yours! Keep in mind to open a draft PR (hopefully a ready one) in under 24 hours 🕐

**Comment by prasoonmhwr** [comment]
@ShubhamPalriwala @jobenjada here is the PR https://github.com/formbricks/formbricks/pull/1037

**Comment by ShubhamPalriwala** [comment]
Hey, this has been fixed while fixing another bug in #1004, hence awarding the points to @rohan9896.

Apologies @prasoonmhwr, please see if you have another issue to work upon or let me know if I can help
